lib/evaluate/call.cc

```diff
@@ -165,18 +165,14 @@ Expr<SubscriptInteger> ProcedureRef::LEN() const {
       const auto *nCopiesArg{
           UnwrapExpr<Expr<SomeInteger>>(arguments_[1].value())};
       CHECK(stringArg != nullptr && nCopiesArg != nullptr);
-      auto stringLen{stringArg->LEN()};
-      return std::move(stringLen) *
-          ConvertTo(stringLen, common::Clone(*nCopiesArg));
-    }
-    if (intrinsic->name == "trim") {
-      // LEN(TRIM(ch)) is unknown without execution.
-      CHECK(arguments_.size() == 1);
-      const auto *stringArg{
-          UnwrapExpr<Expr<SomeCharacter>>(arguments_[0].value())};
-      CHECK(stringArg != nullptr);
-      return stringArg->LEN();
+      if (auto stringLen{stringArg->LEN()}) {
```

Comment:
`stringArg` is of type Expr<SomeType> but I can't seem to find a LEN() associated with this type.
Which LEN() is invoked here?

It's not apparent from the name of the variable `stringLen` or otherwise that it is a pointer. Except that you are dereferencing it later.

Reply:
I'm just trying to understand why `stringLen` wasn't de-referenced before and now it is. Was it a typo?

Reply:
It is `const Expr<SomeCharacter> *`, which should be clear from the declaration above (not part of this change).

Reply:
`stringLen` is now an `optional` so we extract its value with `*` once it is known to be present.